### PR TITLE
[DO_NOT_MERGE] resolve secrets at pod creation time

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/4518 (eventually)

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
[testextended][extended:core(images)]

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

File: pkg/build/util/util.go [hunk 1]
```diff
@@ -167,3 +171,195 @@ func VersionForBuild(build *buildapi.Build) int {
...
+			}
+			return "", err
+		}
+		image := imageStreamImage.Image
+		glog.V(4).Infof("Resolved ImageStreamReference %s to image %s with reference %s in namespace %s", from.Name, image.Name, image.DockerImageReference, namespace)
+		return image.DockerImageReference, nil
+	case "ImageStreamTag":
+		imageStreamTag, err := client.GetImageStreamTag(kapi.WithNamespace(ctx, namespace), from.Name)
+		if err != nil {
+			glog.V(2).Infof("Error resolving ImageStreamTag reference %s in namespace %s: %v", from.Name, namespace, err)
```
Comment (smarterclayton) [inline pkg/build/util/util.go]: Don't log normal errors (tag not found) at v2.

Comment (bparees) [inline pkg/build/util/util.go]: will move it below the existing isNotFound return, but just a heads up, I really only opened this PR to get rosie to do some testing, it's still very much a WIP.

**Pull request closed without merging**